osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -34,7 +33,7 @@ public Strain(int totalColumns)
         protected override double StrainValueOf(DifficultyHitObject current)
         {
             var maniaCurrent = (ManiaDifficultyHitObject)current;
-            var endTime = maniaCurrent.BaseObject.GetEndTime();
+            var endTime = maniaCurrent.EndTime;
```

Comment:
This change (and all the other similar changes in this file) are not actually equivalent. The previous value (read from `BaseObject`) was independent of track rate. This one is not.

Reply:
Ah yea that's totally right. I wonder why I did this in the first place with the original PR 🤔 
It's been a while since I made that change, but I probably thought it was a bug since its counter intuitive that mania calculations would just ignore clockrate like that.

Fixed in ff08eee43b71eeaeb1e16d2d5d8c7f9e321e8899

Out of curiosity, do you know why the mania calc does this?
The whole idea behind clockrate being passed to the DifficultyHitObject abstraction is to remove differences between a clockrate adjusted map and the same objects timing existing as nomod.

Reply:
`ManiaDifficultyCalculator` itself includes adjustments for rate-changing mods.

Reply:
That feels like something that should be addressed.
It very much feels like the difficulty calculator framework was built for osu standard and instead of modifying it to better support the needs of other rulesets, instead other ruleset calculators have just been shoehorned into it and made hacky solutions to fix issues (another example being the taiko `Rhythm` skill's index based strains).
Those are not issues for this PR though.

Reply:
The simple reason why mania does this is that it was build this way from the very beginning. It's not meant to be bug-less, it's meant to replace osu-stable first and foremost to provide a path forward for which such bugs can be fixed and meaningful comparisons can be made against osu-stable.

Reply:
Ah yea that makes sense.
For some reason I thought there had been a major mania restructure after it was ported to lazer. 🤔